src/components/QueuedTransactionList/index.tsx

```diff
@@ -0,0 +1,190 @@
+  account: string,
+  ownedSafes = false,
+): Promise<TransactionSummary[]> => {
+  const { results } = await getTransactionQueue(GATEWAY_URL, chainId, checksumAddress(safeAddress))
+  return results.reduce((acc, cur) => {
+    if (
+      isTransactionSummary(cur) &&
+      isMultisigExecutionInfo(cur.transaction.executionInfo) &&
+      cur.transaction.txStatus === LocalTransactionStatus.AWAITING_CONFIRMATIONS &&
+      (ownedSafes ? cur.transaction.executionInfo.missingSigners?.some(({ value }) => value === account) : true)
+    ) {
+      return [...acc, cur.transaction]
+    }
+    return acc
+  }, [])
+}
+
+type TransactionsSummaryPerChain = {
+  [chainId: ChainId]: {
+    [safeAddress: string]: TransactionSummary[]
+  }
+}
+
+const TxsToConfirmList = (): ReactElement => {
+  const ownedSafes: Record<string, string[]> = useOwnerSafes()
+  const localSafesWithDetails = useLocalSafes()
+  const localSafes: Record<string, string[]> = Object.entries(localSafesWithDetails).reduce(
+    (result, [chain, safes]) => {
+      const safesAddr = safes.map((safe) => safe.address)
+      return { ...result, [chain]: safesAddr }
+    },
+    {},
+  )
+  const userAccount = useSelector(userAccountSelector)
+
+  const [loading, setLoading] = useState<boolean>(true)
+  const [txsAwaitingConfirmation, setTxsAwaitingConfirmation] = useState<TransactionsSummaryPerChain>({})
+  const [displayOwnedSafes, setDisplayOwnedSafes] = useState<boolean>(false)
+  const [isInitialLoad, setIsInitialLoad] = useState<boolean>(true)
+
+  const canDisplayOwnedSafes = displayOwnedSafes && userAccount
+  const safesToTraverse = canDisplayOwnedSafes ? ownedSafes : localSafes
+
+  const title = <h2>Transactions to Sign</h2>
+
+  const handleToggleOwnedSafes = () => {
+    if (userAccount) {
+      setDisplayOwnedSafes((val) => !val)
+      setTxsAwaitingConfirmation({})
+      setIsInitialLoad(true)
+    }
+  }
+
+  // Fetch txs awaiting confirmations after retrieving the owned safes from the LocalStorage
+  useEffect(() => {
+    if (!isInitialLoad || !Object.keys(safesToTraverse || {}).length) {
+      return
+    }
+    const fetchAwaitingConfirmationTxs = async () => {
+      const txs: TransactionsSummaryPerChain = {}
+
+      for (const [chainId, safesPerChain] of Object.entries(safesToTraverse).slice(0, 3)) {
+        txs[chainId] = {}
+        const arrayPromises = safesPerChain.map((safeAddr) => {
+          return getTxsAwaitingConfirmationByChainId(chainId, safeAddr, userAccount, displayOwnedSafes)
+        })
+        const txsByChain = await Promise.all(arrayPromises)
+
+        txsByChain.forEach((summaries, i) => {
+          const safeAddress = safesPerChain[i]
+          txs[chainId][safeAddress] = summaries
+        })
+      }
+      setTxsAwaitingConfirmation(txs)
+      setLoading(false)
+      setIsInitialLoad(false)
+    }
+    fetchAwaitingConfirmationTxs()
+  }, [displayOwnedSafes, userAccount, safesToTraverse])
+
+  if (loading || isInitialLoad) {
```

Comment:
I think you can remove the `loading` state if you decide on an intial loading flag instead.